### ISSUE TITLE
Breakage (due to click update?)

### ISSUE THREAD

**Issue by efroemling** [alanhamlett/pip-update-requirements#37]
Hello,
I've been using pur as part of my build pipeline for a while and in the past few days it has started erroring for me. I'm using Python 3.8 on an intel Mac with everything kept up to date via homebrew and pip

```
cd src && pur -r requirements.txt
Traceback (most recent call last):
  File "/usr/local/bin/pur", line 8, in <module>
    sys.exit(pur())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1134, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1058, in main
    with self.make_context(prog_name, args, **extra) as ctx:
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 927, in make_context
    self.parse_args(ctx, args)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1376, in parse_args
    value, args = param.handle_parse_result(ctx, opts, args)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 2352, in handle_parse_result
    value = self.process_value(ctx, value)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 2308, in process_value
    value = self.type_cast_value(ctx, value)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 2295, in type_cast_value
    return convert(value)
  File "/usr/local/lib/python3.9/site-packages/click/types.py", line 75, in __call__
    return self.convert(value, param, ctx)
  File "/usr/local/lib/python3.9/site-packages/pur/__init__.py", line 54, in convert
    if value.lower() == 'true':
AttributeError: 'bool' object has no attribute 'lower'
```

The versions I'm seeing via pip:
```
pur                      5.4.0
click                    8.0.0
```

It looks like click was just updated to 8.0.0 a few days ago so I'm guessing this broke something?
Please let me know if I can provide any more info..

**Comment by meskio** [comment]
I can confirm the problem comes from the click update, downgrading click to *0.7.2* fixes the problem.

I solve this error by this patch:
```patch
diff --git a/pur/__init__.py b/pur/__init__.py
index d3c32ad..ada8214 100644
--- a/pur/__init__.py
+++ b/pur/__init__.py
@@ -51,7 +51,9 @@ PUR_GLOBAL_UPDATED = 0
 
 class PathOrBool(click.Path):
     def convert(self, value, param, ctx):
-        if value.lower() == 'true':
+        if isinstance(value, bool):
+            return value
+        elif value.lower() == 'true':
             return True
         elif value.lower() == 'false':
             return False
```

But then I got an error somewhere deeper into the code:
```traceback
Traceback (most recent call last):
  File "/home/user/.local/share/virtualenvs/bridgedb/bin/pur", line 11, in <module>
    load_entry_point('pur==5.4.0', 'console_scripts', 'pur')()
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/click/core.py", line 1134, in __call__
    return self.main(*args, **kwargs)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/click/core.py", line 1059, in main
    rv = self.invoke(ctx)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/click/core.py", line 1401, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/click/core.py", line 767, in invoke
    return __callback(*args, **kwargs)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/__init__.py", line 122, in pur
    update_requirements(
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/__init__.py", line 190, in update_requirements
    _internal_update_requirements(obuffer, updates,
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/__init__.py", line 236, in _internal_update_requirements
    for line, req, spec_ver, latest_ver in requirements:
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/__init__.py", line 392, in _get_requirements_and_latest
    latest_ver = latest_version(req, spec_ver, session, finder,
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/utils.py", line 165, in latest_version
    all_candidates = finder.find_all_candidates(req.name)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_internal/index.py", line 610, in find_all_candidates
    for page in self._get_pages(url_locations, project_name):
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_internal/index.py", line 743, in _get_pages
    page = _get_html_page(location, session=self.session)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_internal/index.py", line 229, in _get_html_page
    resp = _get_html_response(url, session=session)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_internal/index.py", line 160, in _get_html_response
    resp = session.get(
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_vendor/requests/sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_internal/download.py", line 403, in request
    return super(PipSession, self).request(method, url, *args, **kwargs)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_vendor/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_vendor/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_vendor/requests/adapters.py", line 416, in send
    self.cert_verify(conn, request.url, verify, cert)
  File "/home/user/.local/share/virtualenvs/bridgedb/lib/python3.9/site-packages/pur-5.4.0-py3.9.egg/pur/packages/pip/_vendor/requests/adapters.py", line 227, in cert_verify
    raise IOError("Could not find a suitable TLS CA certificate bundle, "
OSError: Could not find a suitable TLS CA certificate bundle, invalid path: true
```

I don't have now the time to debug it more.

**Comment by jdhaines** [comment]
I think @meskio typo'd above with the version.  It looks like 7.1.2 was the last version of click before the newest.  I used the following command in my environment to get things running again:
``` bash
pip install -Iv click==7.1.2
```

**Comment by meskio** [comment]
> I think @meskio typo'd above with the version. It looks like 7.1.2 was the last version of click before the newest.

Ups, you are right. 7.1.2 is the version I'm using and it works fine with it.

**Comment by alanhamlett** [comment]
Fixed with 780a30fb50cbb9eddee3b847a8649a57068de9d5 and released in v5.4.1.
